htdocs/user/card.php

```diff
@@ -941,15 +941,19 @@
 		print '<script>
 			jQuery(document).ready(function() {
 				$(".createloginauto").on("change", function(){
-					lastname = $("#lastname").val();
-					firstname = $("#firstname").val();
-					if($(this).attr("id") == "firstname"){
-						firstname = firstname.toLowerCase();
-						firstname = firstname[0];
+					lastname = $("#lastname").val().toLowerCase();
+					firstname = $("#firstname").val().toLowerCase()[0];
+
+					console.log("We generate login intell we have a valid lastname");
+					login = "";
+					if(lastname) {
+						if(firstname) {
+							login = firstname + ".";
```

Comment:
Because the . introduce a behaviour change and may generate trouble for some users, can you replace the 
+ '.' 
with
+ "<php echo getDolGlobalString("MAIN_USER_SEPARATOR_CHAR_FOR_GENERATED_LOGIN"); ?>"
So default doe snot change and end user can decide the separator if he want by hidding the constant MAIN_USER_SEPARATOR_CHAR_FOR_GENERATED_LOGIN to the good char.